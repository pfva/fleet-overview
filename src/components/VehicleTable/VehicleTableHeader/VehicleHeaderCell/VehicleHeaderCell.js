import { useContext } from 'react';
import Icon from '../../../UI/Icon/Icon';
import VehicleContext from '../../../../store/vehicles/vehicle-context';
import styles from './VehicleHeaderCell.module.css';

const VehicleHeaderCell = ({ title, column }) => {
  const vehicleContext = useContext(VehicleContext);

  const sortVehicles = () => {
    console.log(column);
    if (column !== 'equipments' && column !== 'id') {
      vehicleContext.sortAlphabetically(column);
    }
  };

  return (
    <div className={styles.vehicleHeaderWrapper}>
      {column === vehicleContext.sortedColumn && (
        <Icon type="caret-down" className={styles.caret} />
      )}
      <p className={styles.vehicleHeaderCell} onClick={sortVehicles}>
        {title}
      </p>
    </div>
  );
};

export default VehicleHeaderCell;
