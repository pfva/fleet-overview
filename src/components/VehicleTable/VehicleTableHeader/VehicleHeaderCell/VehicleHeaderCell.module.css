.vehicleHeaderCell {
  flex: 1 1 calc(100% / 6);
  margin: 0 var(--spacing-extra-small);
  text-align: left;
  user-select: none;
  cursor: pointer;
}

.vehicleHeaderWrapper {
  flex: 1 1 calc(100% / 6);
  display: flex;
  justify-content: start;
}
