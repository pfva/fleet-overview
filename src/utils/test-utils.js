import React from 'react';
import { render, screen } from '@testing-library/react';
import VehicleProvider from '../store/vehicles/VehicleProvider';
import EquipmentsProvider from '../store/equipments/EquipmentsProvider';

const ProvidersHOC = ({ children }) => {
  return (
    <VehicleProvider>
      <EquipmentsProvider>{children}</EquipmentsProvider>
    </VehicleProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: ProvidersHOC, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
