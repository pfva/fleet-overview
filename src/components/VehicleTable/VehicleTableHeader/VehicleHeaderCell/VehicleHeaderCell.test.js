import { render, screen } from '../../../../utils/test-utils';
import renderer from 'react-test-renderer';
import VehicleHeaderCell from './VehicleHeaderCell';

describe('VehicleHeaderCell', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<VehicleHeaderCell />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the name title', () => {
    render(<VehicleHeaderCell title="name" />);
    const name = screen.getByText(/name/i);
    expect(name).toBeInTheDocument();
  });
});
