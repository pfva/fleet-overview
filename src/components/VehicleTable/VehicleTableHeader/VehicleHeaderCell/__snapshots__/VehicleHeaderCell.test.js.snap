// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleHeaderCell renders without crashing 1`] = `
<div
  className="vehicleHeaderWrapper"
>
  <p
    className="vehicleHeaderCell"
    onClick={[Function]}
  />
</div>
`;
